qiskit/ignis/verification/quantum_volume/fitters.py

```diff
@@ -256,58 +256,68 @@ def calc_statistics(self):
                                         (1.0-self._ydata[2][depthidx])
                                         / self._ntrials)**0.5
 
-    def plot_qv_data(self, ax=None, show_plt=True):
+    def plot_qv_data(self, ax=None, show_plt=True, figsize=(7, 5)):
         """
         Plot the qv data as a function of depth
 
         Args:
             ax (Axes or None): plot axis (if passed in).
             show_plt (bool): display the plot.
+            figsize (tuple): Figure size in inches.
 
         Raises:
             ImportError: If matplotlib is not installed.
+
+        Returns:
+            matplotlib.Figure:
+                A figure of Quantum Volume data (heavy output probability)
+                with two-sigma error bar as a function of circuit depth
```

Comment:
The doc failure is because of this, the syntax here is wrong, without text on the line after the semicolon the sphinx rst parser can't distinguish this between a header and the return type. This should be like:

```suggestion
            matplotlib.Figure: A figure of Quantum Volume data (heavy
                output probability) with two-sigma error bar as a function
                of circuit depth
```

(although that might have other linter errors because github's code editor/suggestions interface is pretty bad)

Reply:
Making this changes doesn't fix the doc failure. I follow the format of `plot_histogram` which I assume passes the doc test. https://github.com/Qiskit/qiskit-terra/blob/e8aa1daf5eaa9467357e482e3172f1d384303ec0/qiskit/visualization/counts_visualization.py#L75

Reply:
That was my bad I misread the error message and the formatting you used here was quite weird so I assumed it was that. Adding more verbosity to to the sphinx build output with `tox -edocs -- -vvvv` on this PR locally returns:

```
[autodoc] from qiskit.ignis.verification import QVFitter.plot_qv_data
[autodoc] import qiskit.ignis.verification => <module 'qiskit.ignis.verification' from '/home/mtreinish/git/qiskit/qiskit-ignis/qiskit/ignis/verification/__init__.py'>
[autodoc] getattr(_, 'QVFitter')
[autodoc] => <class 'qiskit.ignis.verification.quantum_volume.fitters.QVFitter'>
[autodoc] getattr(_, 'plot_qv_data')
[autodoc] => <function QVFitter.plot_qv_data at 0x7f80640dd940>
[app] emitting event: 'autodoc-before-process-signature'(<function QVFitter.plot_qv_data at 0x7f80640dd940>, True)
[app] emitting event: 'autodoc-process-signature'('method', 'qiskit.ignis.verification.QVFitter.plot_qv_data', <function QVFitter.plot_qv_data at 0x7
[app] emitting event: 'autodoc-process-docstring'('method', 'qiskit.ignis.verification.QVFitter.plot_qv_data', <function QVFitter.plot_qv_data at 0x7
[app] emitting event: 'build-finished'(ExtensionError("Handler <function process_docstring at 0x7f807a5c0430> for event 'autodoc-process-d

Traceback (most recent call last):
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/events.py", line 110, in emit
    results.append(listener.handler(self.app, *args))
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx_autodoc_typehints.py", line 364, in process_docstring
    type_hints = get_all_type_hints(obj, name)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx_autodoc_typehints.py", line 219, in get_all_type_hints
    rv = backfill_type_hints(obj, name)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx_autodoc_typehints.py", line 260, in backfill_type_hints
    obj_ast = ast.parse(textwrap.dedent(inspect.getsource(obj)), **parse_kwargs)
  File "/usr/lib/python3.8/ast.py", line 47, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 1
    def plot_qv_data(self, ax=None, show_plt=True, figsize=(7, 5)):
    ^
IndentationError: unexpected indent

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/cmd/build.py", line 280, in build_main
    app.build(args.force_all, filenames)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/application.py", line 348, in build
    self.builder.build_update()
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 297, in build_update
    self.build(to_build,
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 311, in build
    updated_docnames = set(self.read())
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 418, in read
    self._read_serial(docnames)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 439, in _read_serial
    self.read_doc(docname)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 479, in read_doc
    doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/io.py", line 223, in read_doc
    pub.publish()
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/core.py", line 217, in publish
    self.document = self.reader.read(self.source, self.parser,
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/io.py", line 128, in read
    self.parse()
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/readers/__init__.py", line 77, in parse
    self.parser.parse(self.input, document)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/parsers.py", line 102, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 170, in run
    results = StateMachineWS.run(self, input_lines, input_offset,
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2769, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 393, in new_subsection
    newabsoffset = self.nested_parse(
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2344, in explicit_markup
    self.explicit_list(blank_finish)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2369, in explicit_list
    newline_offset, blank_finish = self.nested_list_parse(
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 318, in nested_list_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2647, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2354, in explicit_construct
    return method(self, expmatch)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2096, in directive
    return self.run_directive(
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2146, in run_directive
    result = directive_instance.run()
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/ext/autodoc/directive.py", line 157, in run
    result = parse_generated_content(self.state, params.result, documenter)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/ext/autodoc/directive.py", line 104, in parse_generated_content
    state.nested_parse(content, 0, node)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2342, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2354, in explicit_construct
    return method(self, expmatch)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2096, in directive
    return self.run_directive(
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2146, in run_directive
    result = directive_instance.run()
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/domains/__init__.py", line 285, in run
    return super().run()
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/directives/__init__.py", line 209, in run
    self.state.nested_parse(self.content, self.content_offset, contentnode)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2344, in explicit_markup
    self.explicit_list(blank_finish)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2369, in explicit_list
    newline_offset, blank_finish = self.nested_list_parse(
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 318, in nested_list_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2647, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2354, in explicit_construct
    return method(self, expmatch)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2096, in directive
    return self.run_directive(
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2146, in run_directive
    result = directive_instance.run()
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/ext/autosummary/__init__.py", line 246, in run
    items = self.get_items(names)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/ext/autosummary/__init__.py", line 384, in get_items
    documenter.add_content(None)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/ext/autodoc/__init__.py", line 567, in add_content
    for i, line in enumerate(self.process_doc(docstrings)):
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/ext/autodoc/__init__.py", line 532, in process_doc
    self.env.app.emit('autodoc-process-docstring',
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/application.py", line 450, in emit 
    return self.events.emit(event, *args, allowed_exceptions=allowed_exceptions)
  File "/home/mtreinish/git/qiskit/qiskit-ignis/.tox/docs/lib/python3.8/site-packages/sphinx/events.py", line 117, in emit 
    raise ExtensionError(__("Handler %r for event %r threw an exception") %
sphinx.errors.ExtensionError: Handler <function process_docstring at 0x7f807a5c0430> for event 'autodoc-process-docstring' threw an exception (exception: unexpected indent (<unknown>, line 1))
```

Which I had never seen before. Doing a bit of research I found https://github.com/agronholm/sphinx-autodoc-typehints/issues/139 which gave me enough background info to identify the real problem. It's the dedent on your fstring confusing the parser. If you change it to be like:
```diff
diff --git a/qiskit/ignis/verification/quantum_volume/fitters.py b/qiskit/ignis/verification/quantum_volume/fitters.py
index 9ab45b5..1552f62 100644
--- a/qiskit/ignis/verification/quantum_volume/fitters.py
+++ b/qiskit/ignis/verification/quantum_volume/fitters.py
@@ -310,8 +310,9 @@ class QVFitter:
         ax.grid(True)
 
         ax.legend()
-        ax.set_title(f'Quantum Volume for up to {len(self._qubit_lists[-1])} Qubits \
-and {self._ntrials} Trials')
+        ax.set_title(
+            f'Quantum Volume for up to {len(self._qubit_lists[-1])} Qubits and '
+            f'{self._ntrials} Trials')
         plt.close(fig)  # close additional figure
 
         if show_plt:
```

Reply:
Wow thanks so much for figuring out the real problem. I am going to fix it right now.